adding application launcher issue #145

akoserwal opened the pull request:
https://github.com/patternfly/patternfly-ng/issues/145
I need to work on example and adding test case

dlabrecq committed:
feat(table): added datatable implementation

AdamJ committed:
fix(datatable): Update datatable to use PatternFly table styles

Update the datatable to use the PatternFly table styles. This includes hover, selection, drag and drop and the toolbar.

dlabrecq committed:
chore(datatable): added expanding rows, de/select all rows, more examples, etc

dlabrecq committed:
chore(release): updated patternfly-eng-release version

Changed patternfly-eng-release script to not run build:demo as the build already runs this. Travis will complete much faster if we don’t run the build twice.

akoserwal committed:
adding application launcher

akoserwal committed:
adding application launcher

dlabrecq commented on src/app/navigation/application-launcher.component.ts:
```diff
@@ -0,0 +1,66 @@
+ * Application Launcher component
+ * @param {boolean=} isDisabled Disable the application launcher button, default: false
+ * @param {boolean=} isList Display items as a list instead of a grid, default: false
+ * @param {boolean=} hiddenIcons Flag to not show icons on the launcher, default: false
```
The inputs are documented below (the docs will pick that up), so it's not necessary to add that here.

akoserwal replied:
@dlabrecq : sure, I will remove it.

dlabrecq commented on src/app/navigation/application-launcher.component.ts:
```diff
@@ -0,0 +1,66 @@
+@Component({
+    encapsulation: ViewEncapsulation.None,
+    selector: 'pfng-application-launcher',
+    styleUrls: ['./application-launcher.component.less'],
```
I don't see a less file. Did you forget to commit that?

akoserwal replied:
@dlabrecq: yeah, I forget to commit it.

dlabrecq replied:
Please remove the styleUrls property.

To allow for customization, we no longer embed the CSS in the component. Instead, we now build the CSS as a single file, under the src/assets/stylesheets directory.

This change was introduced last week -- please pull the latest.

dlabrecq reviewed (changes requested):
Please move this component under its own directory -- similar to vertical-naviagtion. 

Will need an example and unit tests before we can review further. A link to a working example and screen shots will also be helpful.

dlabrecq committed:
fix(release): remove duplicate c3 package

dlabrecq committed:
chore(table): cleanup docs

akoserwal commented:
In angular-patternfly, application launcher was kind of part of navigation directory. i.e why keep it that way. sure, I will move it into separate. yes, I am working on the example & test case part. thanks

andrewazores committed:
fix(sparkline): avoid regenerating chart on every data update

When only chart data has changed and config remains unchanged, the sparkline chart component will
not be regenerated but the data simply reloaded instead. This should be a lighter-weight operation
and avoids a visual flicker and redraw of the chart component on data update.

fixes https://github.com/patternfly/patternfly-ng/issues/275

dlabrecq commented:
We recently cleaned up some things, so navigation/application-launcher would be fine.

dlabrecq committed:
Merge pull request #273 from dlabrecq/c3

Remove duplicate c3 package

dlabrecq committed:
Merge pull request #274 from dlabrecq/table-cleanup

Table: minor doc cleanup

dlabrecq committed:
chore(table): added doc comments to table component

dlabrecq committed:
fix(donut): avoid regenerating chart on every data update

This smooths out the animation when transitioning between data sets

dlabrecq committed:
Merge pull request #276 from andrewazores/master

fix(sparkline): avoid regenerating chart on every data update

dlabrecq committed:
Merge pull request #277 from dlabrecq/table-doc

Table documentation

dlabrecq committed:
chore(pagination): fix dropdown test for compatibility with ngx-bootstrap 1.8.1 and above

dlabrecq committed:
chore(tests): update test dependencies

Although the latest dependencies are not required to fix the dropdown tests with ngx-bootstrap 1.8.1 and above; testing with the latest appears solid.

dlabrecq committed:
chore(sort): fix dropdown test for compatibility with ngx-bootstrap 1.8.1 and above

dlabrecq committed:
chore(action): fix dropdown test for compatibility with ngx-bootstrap 1.8.1 and above

dlabrecq committed:
chore(filter): fix dropdown test for compatibility with ngx-bootstrap 1.8.1 and above

dlabrecq committed:
chore(card): fix dropdown test for compatibility with ngx-bootstrap 1.8.1 and above

dlabrecq committed:
chore(tree-list): update comments and run at least one test

dlabrecq committed:
chore(table): fix dropdown test for compatibility with ngx-bootstrap 1.8.1 and above

dlabrecq committed:
chore(toolbar): fix dropdown test for compatibility with ngx-bootstrap 1.8.1 and above

seanforyou23 committed:
chore(tests): remove unnecessary statements in unit tests for sort component

when relying on tick to simulate passage of time, we only need
to call fixture.detectChanges() after the tick() statement

dlabrecq committed:
chore(test): use headless chrome instead of PhantomJS to enable tree-list tests

Also fixed errors generated during test cleanup because vertical nav listeners were not destroyed properly

seanforyou23 committed:
chore(tests): remove unnecessary statements in unit test

for toolbar component

when relying on tick to simulate passage of time, we only need
to call fixture.detectChanges() after the tick() statement

seanforyou23 committed:
chore(tests): remove unnecessary statements in unit test

for action component

when relying on tick to simulate passage of time, we only need
to call fixture.detectChanges() after the tick() statement

seanforyou23 committed:
chore(tests): remove unnecessary statements in unit test

for card component

when relying on tick to simulate passage of time, we only need
to call fixture.detectChanges() after the tick() statement

seanforyou23 committed:
chore(filter-tests): remove unnecessary statements in unit test

when relying on tick to simulate passage of time, we only need
to call fixture.detectChanges() after the tick() statement

seanforyou23 committed:
chore(pagination-tests): remove unnecessary statements in unit test

when relying on tick to simulate passage of time, we only need
to call fixture.detectChanges() after the tick() statement

tested with ngx-bootstrap v1.8.0 and 1.9.3

seanforyou23 committed:
chore(table-tests): remove unnecessary statements in unit test

when relying on tick to simulate passage of time, we only need
to call fixture.detectChanges() after the tick() statement

tested with ngx-bootstrap v1.8.0 and 1.9.3

dlabrecq committed:
Merge pull request #298 from seanforyou23/table-tests-cleanup

chore(table-tests): remove unnecessary statements in unit test

dlabrecq committed:
Merge pull request #297 from seanforyou23/pagination-tests-cleanup

chore(pagination-tests): remove unnecessary statements in unit test

dlabrecq committed:
Merge pull request #296 from seanforyou23/filter-tests-cleanup

chore(filter-tests): remove unnecessary statements in unit test

dlabrecq committed:
Merge pull request #295 from seanforyou23/card-tests-cleanup

chore(card-tests): remove unnecessary statements in unit test

dlabrecq committed:
Merge pull request #294 from seanforyou23/action-tests-cleanup

chore(action-tests): remove unnecessary statements in unit test

dlabrecq committed:
Merge pull request #292 from seanforyou23/tests-cleanup

chore(sort-tests): remove unnecessary statements in unit test

dlabrecq committed:
Merge pull request #293 from seanforyou23/toolbar-tests-cleanup

chore(toolbar-tests): remove unnecessary statements in unit test

dlabrecq committed:
feat(modules): each module should have its own index.ts

This allows patternfly-ng modules to be imported like so:

import { NotificationModule } from 'patternfly-ng/notification';

dlabrecq committed:
feat(css): remove embedded css to allow customization

BREAKING CHANGE: Apps must now include patternfly-ng.min.css

akoserwal commented:
@dlabrecq : Will application-launcher will be part of navigation module? same as vertical component.

seanforyou23 committed:
chore(deps): use more flexible range for ngx-bootstrap dependency version

dlabrecq committed:
Merge pull request #305 from seanforyou23/incremental-ngx-bs-upgrade

chore(deps): use flexible range for ngx-bootstrap dep version

dlabrecq commented:
Yes, just under a subdirectory -- src/app/navigation/application-launcher

dlabrecq committed:
perf(tree-list): bumped angular-tree-component to v7.0.0

dlabrecq committed:
chore(readme): add doc about optional dependencies

dlabrecq committed:
Merge pull request #308 from dlabrecq/optional-docs

Add doc about optional dependencies

akoserwal committed:
updated launcher component and template

akoserwal committed:
updated launcher component and template

akoserwal committed:
added app launcher spec and less file

akoserwal committed:
added app launcher spec and less file

akoserwal committed:
app launcher example and updated nav example module

akoserwal committed:
app launcher example and updated nav example module

akoserwal committed:
updating routing for nav module and adding app launcher to navbar

akoserwal committed:
updating routing for nav module and adding app launcher to navbar

LHinson committed:
Update CONTRIBUTING.md

LHinson committed:
Update CONTRIBUTING.md

akoserwal committed:
working app launcher

akoserwal committed:
working app launcher

dlabrecq commented on src/app/navigation/examples/application-launcher-example.component.html:
```diff
@@ -0,0 +1,61 @@
+<ul class="nav navbar-nav">
```
Please rename the 'examples' directory as 'example'.

Also, please move the component and example under a directory, named src/app/navigation/application-launcher

dlabrecq commented on index.ts:
```diff
@@ -113,6 +113,7 @@ export { WindowReference } from './src/app/utilities/window.reference';
 export { NavigationItemConfig } from './src/app/navigation/navigation-item-config';
 export { NavigationModule } from './src/app/navigation/navigation.module';
 export { VerticalNavigationComponent } from './src/app/navigation/vertical-navigation.component';
+export { ApplicationLauncherComponent} from './src/app/navigation/application-launcher.component';
```
There is now a new index.ts file in each component directory. This allows a dev to import individual modules like so:

import { NavigationModule } from 'patternfly-ng/navigation';

Please add this export to src/app/navigation/index.js -- you'll need to pull the latest source.

dlabrecq commented on src/demo/navbar/navbar-items.ts:
```diff
@@ -65,8 +65,16 @@ export class NavbarItems {
+    children: [{
+      id: 'verticalnavigation',
+      path: 'verticalnavigation',
+      title: 'Vertical Navigation Component',
```
This should just be labeled "Vertical Navigation"

dlabrecq commented on src/app/navigation/examples/application-launcher-example.component.html:
```diff
@@ -0,0 +1,61 @@
+      </a>
+      <ul class="dropdown-menu" aria-labelledby="dropdownMenu2">
+        <li><a href="#">Preferences</a></li>
+        <li><a href="#">Logout</a></li>
```
These drop downs don't appear to function. For example, they don't open to show options. Is that intentional?

dlabrecq commented on src/app/navigation/examples/application-launcher-example.component.html:
```diff
@@ -0,0 +1,61 @@
+        [isList]="false"
+        [items]="navigationItems"
+        [hiddenIcons] = "false">
+      </pfng-application-launcher>
```
![screen shot 2018-01-25 at 5 40 34 pm](https://user-images.githubusercontent.com/17481322/35416421-08a47c76-01f7-11e8-979b-a20c3808c08a.png)
The app launcher doesn't appear aligned with the other dropdowns.

dlabrecq commented on src/app/navigation/examples/application-launcher-example.component.html:
```diff
@@ -0,0 +1,61 @@
+<ul class="nav navbar-nav">
+</ul>
+<ul class="nav navbar-nav navbar-right navbar-iconic">
+    <li class="dropdown">
```
Clicking on the items in the app launcher opens a new window and leaves the demo in a funky state -- the welcome page is blank.

It appears to be related to the Url used to launch the new window. The error in the browser console is:

Error: Uncaught (in promise): Error: Cannot match any routes. URL Segment: 'ipsum/intellegam/suavitate'

![screen shot 2018-01-25 at 5 44 46 pm](https://user-images.githubusercontent.com/17481322/35416546-791f5714-01f7-11e8-9f7f-766c39bd499c.png)

dlabrecq commented on src/app/navigation/application-launcher.component.ts:
```diff
@@ -0,0 +1,72 @@
+      this.isClicked = false;
+    }
+
+    toggle() {
```
Please make this private if possible. Otherwise, it needs docs

dlabrecq commented on src/app/navigation/application-launcher.component.ts:
```diff
@@ -0,0 +1,72 @@
+
+    }
+
+    ngOnInit(): void {
```
Needs doc

dlabrecq commented on src/app/navigation/application-launcher.component.ts:
```diff
@@ -0,0 +1,72 @@
+    templateUrl: './application-launcher.component.html'
+})
+
+export class ApplicationLauncherComponent  implements OnInit {
```
The ApplicationLauncherComponent class needs some simple doc. Each class and public method shows up in the TS documentation.

dlabrecq commented on src/app/navigation/application-launcher.component.spec.ts:
```diff
@@ -0,0 +1,11 @@
+  
+  import { NavigationItemConfig } from './navigation-item-config';
+  import { TooltipModule } from 'ngx-bootstrap';
+  import { ApplicationLauncherComponent } from './application-launcher.component';
```
Please add some tests

dlabrecq reviewed (changes requested):
Reviewed the latest commits...

akoserwal committed:
updating routing for launcher

akoserwal committed:
updated app launcher doc, added test case

akoserwal committed:
added app launcher

akoserwal committed:
added app launcher to nav module and removed from vertical nav example module

dlabrecq commented:
Closing in favor of PR #310.
